_posts/2024-03-20-kraft-migration.md

```diff
@@ -0,0 +1,201 @@
+---
+layout: post
+title:  "From ZooKeeper to KRaft: how the Kafka migration works"
+date: 2024-03-20
+author: paolo_patierno
+---
+
+The Apache Kafka project has been using [Apache ZooKeeper](https://zookeeper.apache.org/) since its inception to store metadata.
+Registered brokers, the current controller and topics' configuration are just a few of the data that ZooKeeper stores for supporting a running Kafka cluster.
+Using a dedicated and centralized system for maintaining cluster metadata and offload leader election was the right approach at that time, because ZooKeeper was battle tested and it is purpose built for providing distributed coordination.
+But taking care of an additional component alongside your Kafka cluster is not that simple.
+Furthermore, ZooKeeper has become a bottleneck that limits the amount of partitions that a single broker can handle.
+For these and other reasons, the Kafka community opened the "famous" [KIP-500](https://cwiki.apache.org/confluence/display/KAFKA/KIP-500%3A+Replace+ZooKeeper+with+a+Self-Managed+Metadata+Quorum) in late 2019 and started to work on the ZooKeeper removal.
```

Comment:
From a reader's POV, I'd think this article is talking about the "how the migration works" from the article title, but from the first paragraph it looks like we didn't mention anything about the migration. Should we at least say something about it here?